java/src/main/java/com/google/appengine/tools/mapreduce/MapReduceJob.java

```diff
@@ -204,6 +241,22 @@ private static class SortJob extends Job1<
     @NonNull private final MapReduceSpecification<?, ?, ?, ?, ?> mrSpec;
     @NonNull private final MapReduceSettings settings;
 
+    @Setter(onMethod = @__(@VisibleForTesting))
+    private transient Datastore datastore;
+
+    @Setter(onMethod = @__(@VisibleForTesting))
+    @Inject
+    transient ShardedJobService shardedJobService;
+
+    protected Datastore getDatastore() {
+      if (datastore == null) {
+        datastore = DatastoreOptions.getDefaultInstance().toBuilder()
```

Comment:
why can't you do ` datastore = settings.getDatastoreOptions().getService();` like in the MapJob?